src/Animations/babylon.animation.ts

```diff
@@ -529,14 +529,27 @@
 
             for (index = 0; index < parsedAnimation.keys.length; index++) {
                 var key = parsedAnimation.keys[index];
-
+                var inTangent:any;
+                var outTangent:any;
 
                 switch (dataType) {
                     case Animation.ANIMATIONTYPE_FLOAT:
                         data = key.values[0];
+                        if (key.values.length >= 1) {
+                            inTangent = key.values[1];
+                        }
+                        if (key.values.length >= 2) {
+                            outTangent = key.values[2];
+                        }
                         break;
                     case Animation.ANIMATIONTYPE_QUATERNION:
                         data = Quaternion.FromArray(key.values);
+                        if (key.values.length >= 8) {

```

Comment:
same here. You cannot be sure the animation was not meant to be linear

Reply:
fixed at 2b8c9e0.
if the rotation Quaternion animation at this time is designed for linear, the Babylon.js engine doesn't use 'quaternionInterpolateFunctionWithTangents'
Did you mean it?

Reply:
My question is more about not loading unwanted tangents

Reply:
OK your feedback is that?

In case of linear animation, the engine load this style
```
"keys" : [
{
	"frame" : 0,
	"values" : [50.673542]
},
{
	"frame" : 15,
	"values" : [-3.11644864]
},
{
	"frame" : 25,
	"values" : [-36.6116676]
}
}
```
in case of tangent interpolation, load this style
```
"keys" : [
{
	"frame" : 0,
	"values" : [50.673542, -3.38392878, -3.38392878]
},
{
	"frame" : 15,
	"values" : [-3.11644864, -3.572852, -3.572852]
},
{
	"frame" : 25,
	"values" : [-36.6116676, -3.10995674, -3.10995674]
}
}
```
※ it's float animation case.

Reply:
Really true :)